modules/core/include/opencv2/core/eigensystem.hpp

```diff
@@ -0,0 +1,674 @@
+#ifndef __EIGENSYSTEM_HPP__
+#define __EIGENSYSTEM_HPP__
+
+using namespace std;
+using namespace cv;
+/**
+ *
+ * This class is just a rip-off of the EigenvalueSolver in JAMA
+ * (http://math.nist.gov/javanumerics/jama), which include public-domain
+ * implementation of linear algebra algorithms. I give the credit to the
+ * original authors (http://math.nist.gov/javanumerics/jama/#Authors).
+ *
+ * Copyright Notice in JAMA
+ *
+ * This software is a cooperative product of The MathWorks and
+ * the National Institute of Standards and Technology (NIST) which has been
+ * released to the public domain. Neither The MathWorks nor NIST assumes any
+ * responsibility whatsoever for its use by other parties, and makes no
+ * guarantees, expressed or implied, about its quality, reliability, or any
+ * other characteristic.
+ *
+ */
+
+class EigenvalueDecomposition {
+private:
+  int n;
+  double cdivr, cdivi;
+
+  double *d, *e, *ort;
+  double **V, **H;
+
+  template <typename _Tp> _Tp *alloc_1d(int m) { return new _Tp[m]; }
+
+  template <typename _Tp> _Tp *alloc_1d(int m, _Tp val) {
+    _Tp *arr = alloc_1d<_Tp>(m);
+    for (int i = 0; i < m; i++)
+      arr[i] = val;
+    return arr;
+  }
+
+  template <typename _Tp> _Tp **alloc_2d(int m, int nn) {
+    _Tp **arr = new _Tp *[m];
+    for (int i = 0; i < m; i++)
+      arr[i] = new _Tp[nn];
+    return arr;
+  }
+
+  template <typename _Tp> _Tp **alloc_2d(int m, int nn, _Tp val) {
+    _Tp **arr = alloc_2d<_Tp>(m, nn);
+    for (int i = 0; i < m; i++) {
+      for (int j = 0; j < nn; j++) {
+        arr[i][j] = val;
+      }
+    }
+    return arr;
+  }
+
+  void cdiv(double xr, double xi, double yr, double yi) {
+    double r, dd;
+    if (abs(yr) > abs(yi)) {
+      r = yi / yr;
+      dd = yr + r * yi;
+      cdivr = (xr + r * xi) / dd;
+      cdivi = (xi - r * xr) / dd;
+    } else {
+      r = yr / yi;
+      dd = yi + r * yr;
+      cdivr = (r * xr + xi) / dd;
+      cdivi = (r * xi - xr) / dd;
+    }
+  }
+
+  // Nonsymmetric reduction from Hessenberg to real Schur form.
+
+  void hqr2() {
+
+    //  This is derived from the Algol procedure hqr2,
+    //  by Martin and Wilkinson, Handbook for Auto. Comp.,
+    //  Vol.ii-Linear Algebra, and the corresponding
+    //  Fortran subroutine in EISPACK.
```

Comment:
Please avoid duplication of code and reuse/fix existed one:
https://github.com/opencv/opencv/blob/3.3.0/modules/core/src/lda.cpp#L319-L322

Reply:
Is there an appropriate include statement to access the functions in ida.cpp?